perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_capabilities.java

```diff
@@ -0,0 +1,57 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.Resource;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.ResourceAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.ResourceAttributesModuleImplApi;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Module for capabilities according to AARC specification. i.e. specification of resource and optional actions.
+ * Syntax:
+ * value=“res:RESOURCE[:CHILD-RESOURCE1][:CHILD-RESOURCE2]...[:act:ACTION[,ACTION]...]”
+ * Example1: value=“res:RESOURCE1:CHILD-RESOURCE1:CHILD-RESOURCE2:act:ACTION1,ACTION2,ACTION3”
+ * Example2: value=“res:RESOURCE1:CHILD-RESOURCE1:CHILD-RESOURCE2”
+ * Example3: value=“res:=$:(:)”
+ *
+ * @author Ondrej Ernst
+ */
+public class urn_perun_resource_attribute_def_def_capabilities extends ResourceAttributesModuleAbstract implements ResourceAttributesModuleImplApi {
+
+	private static final String resourceAllowedCharacters = "[a-zA-Z0-9()+,\\-.=@;$_!*']";
+	private static final String actionAllowedCharacters = "[a-zA-Z0-9()+\\-.:=@;$_!*']";
+	private static final Pattern pattern = Pattern.compile("res((?!(:act$|:act:|.*:,|.*,:|.*::|.*:$))(:" + // what the whole string cannot contain
+		resourceAllowedCharacters + "+))+(:act:((" + actionAllowedCharacters +
+		"+,)*(" + actionAllowedCharacters + "+)))?");
+
+
+	@Override
+	public void checkAttributeSyntax(PerunSessionImpl perunSession, Resource resource, Attribute attribute) throws InternalErrorException, WrongAttributeValueException {
+		if (attribute.getValue() == null) return;
+		List<String> values = attribute.valueAsList();
+		for (String value : values) {
+			if (!pattern.matcher(value).matches()) {
+				throw new WrongAttributeValueException(attribute, resource.getName() + " has attribute whose value is not valid");
```

Comment:
Add there an example of valid value
`res:RESOURCE[:CHILD-RESOURCE1][:CHILD-RESOURCE2]...[:act:ACTION[,ACTION]...]`